### PR TITLE
GEODE-7917: change thrown exception type for SSL to IOException

### PR DESCRIPTION
**Pull request by mivanac**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [*] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [*] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [*] Is your initial contribution a single, squashed commit?

- [*] Does `gradlew build` run cleanly?

- [x] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-core/src/integrationTest/java/org/apache/geode/distributed/internal/tcpserver/TCPClientSSLIntegrationTest.java [hunk 1]
```diff
@@ -77,6 +80,26 @@ public void after() {
...
+    serverStore.withCertificate("server", serverCertificate);
+    serverStore.trust("ca", ca);
+
+    Properties serverProperties = serverStore
+        .propertiesWith(LOCATOR, true, true);
+
+    startTcpServer(serverProperties);
+  }
+
+
```
Comment (mkevo) [inline geode-core/src/integrationTest/java/org/apache/geode/distributed/internal/tcpserver/TCPClientSSLIntegrationTest.java:102]: Please remove one blank line.

File: geode-core/src/integrationTest/java/org/apache/geode/distributed/internal/tcpserver/TCPClientSSLIntegrationTest.java [hunk 2]
```diff
@@ -216,4 +239,37 @@ public void clientFailsToConnectIfServerCertificateWrongHostname() throws Except
             + localhost.getHostName() + " found.");
   }
 
+  @Test
+  public void clientFailsToConnectIfRemotePeerShutdowns() throws Exception, SSLHandshakeException {
+
+    startServerWithCertificate();
+
+
```
Comment (mkevo) [inline geode-core/src/integrationTest/java/org/apache/geode/distributed/internal/tcpserver/TCPClientSSLIntegrationTest.java:247]: The same comment as above.

File: geode-core/src/integrationTest/java/org/apache/geode/distributed/internal/tcpserver/TCPClientSSLIntegrationTest.java [hunk 3]
```diff
@@ -216,4 +239,37 @@ public void clientFailsToConnectIfServerCertificateWrongHostname() throws Except
...
+    SocketCreator socketCreator = Mockito.mock(SocketCreator.class);
+    ClusterSocketCreator ssc = Mockito.mock(ClusterSocketCreator.class);
+
+
```
Comment (mkevo) [inline geode-core/src/integrationTest/java/org/apache/geode/distributed/internal/tcpserver/TCPClientSSLIntegrationTest.java:251]: The same.

File: geode-core/src/integrationTest/java/org/apache/geode/distributed/internal/tcpserver/TCPClientSSLIntegrationTest.java [hunk 4]
```diff
@@ -216,4 +239,37 @@ public void clientFailsToConnectIfServerCertificateWrongHostname() throws Except
...
+        InternalDataSerializer.getDSFIDSerializer().getObjectSerializer(),
+        InternalDataSerializer.getDSFIDSerializer().getObjectDeserializer(),
+        TcpSocketFactory.DEFAULT);
+
+    assertThatExceptionOfType(IOException.class)
+        .isThrownBy(() -> client.requestToServer(new HostAndPort(localhost.getHostName(), port),
+            Boolean.valueOf(false), 5 * 1000))
+        .withCauseInstanceOf(SSLHandshakeException.class)
+        .withStackTraceContaining("Remote host terminated the handshake");
+
```
Comment (mkevo) [inline geode-core/src/integrationTest/java/org/apache/geode/distributed/internal/tcpserver/TCPClientSSLIntegrationTest.java:272]: Remove this line.

File: geode-tcp-server/src/main/java/org/apache/geode/distributed/internal/tcpserver/TcpClient.java [hunk 5]
```diff
@@ -255,6 +256,14 @@ private Short getServerVersion(HostAndPort addr, int timeout)
...
       sock = socketCreator.forCluster().connect(addr, timeout, null, socketFactory);
       sock.setSoTimeout(timeout);
+    } catch (SSLHandshakeException e) {
+      if ((e.getCause() instanceof EOFException)
+          && (e.getCause().getMessage().contains("SSL peer shut down incorrectly"))) {
+        throw new IOException("Remote host terminated the handshake", e);
+      } else {
+        throw new IllegalStateException("Unable to form SSL connection", e);
+      }
+
```
Comment (mkevo) [inline geode-tcp-server/src/main/java/org/apache/geode/distributed/internal/tcpserver/TcpClient.java:266]: Also remove this. Thanks.

Comment (mivanac) [reply geode-tcp-server/src/main/java/org/apache/geode/distributed/internal/tcpserver/TcpClient.java]: Thanks for comments.

**Pull request merged**